### PR TITLE
Replacing the Paket.Core with Nuget.Protocol library

### PR DESCRIPTION
**Pull request by WojcikMike**

### CODE REVIEW

File: src/Common/Runner/Nuget/NuGetPackageResolver.cs [hunk 1]
```diff
@@ -20,38 +23,45 @@ public NuGetPackageResolver(string packagesStore, string nugetFeed = Constants.D
...
+            var packageLocation = await DownloadPackage(packageInfo, latesVersion);
+            var dependencies = await GetAllDependencies(packageInfo, latesVersion);
 
             var package = new Package
             {
                 Info = packageInfo,
-                Version = latesVersion,
-                Files = files.Where(f => Path.GetExtension(f.Name) == ".dll").ToArray()
+                Version = latesVersion.ToString(),
+                Files = dependencies.Where(f => Path.GetExtension(f.Name) == ".dll").ToArray()
```
Comment (timbussmann) [inline src/Common/Runner/Nuget/NuGetPackageResolver.cs:34]: is that the same as the files contained before? it looks like files previously was only referring to files of a specific package while at first glance, it now looks like it contains fails of dependencies too?

File: src/Common/Runner/Nuget/NuGetPackageResolver.cs [hunk 2]
```diff
@@ -63,39 +73,82 @@ async Task<IEnumerable<FileInfo>> GetAllDependencies(PackageInfo package, SemVer
...
+                    cache,
+                    logger,
+                    cancellationToken);
+
+                Console.WriteLine($"Downloaded package {package.PackageName} {version}");
+
+                using (var packageReader = new PackageArchiveReader(packageStream))
+                {
+
+                    return packageReader.GetFiles().First();
```
Comment (timbussmann) [inline src/Common/Runner/Nuget/NuGetPackageResolver.cs:101]: what is the code returning here?

**Comment by timbussmann** [comment]
I think we can close this one @WojcikMike ?

**Comment by WojcikMike** [comment]
Yep.

**Pull request closed without merging**